libnethack/src/rtp.c

```diff
@@ -0,0 +1,108 @@
+#include "rtp.h"
+
+struct monst *
+name_rtp(struct monst *mtmp)
+{
+    const char* rtpNames[RTP_NAME_COUNT] = RTP_NAMES;
+    int rtp_id = rn2(RTP_NAME_COUNT);
+
+    if (FEMALE_RTPS(rtp_id)) {
+        mtmp->female = TRUE;
+    }
+
+    return christen_monst(mtmp, msg_from_string(rtpNames[rtp_id]));
+}
+
+struct obj *
+create_rtp_corpse(struct level *lev, int x, int y, enum rng rng)
+{
+    struct obj *obj = NULL;
+    struct obj *orig_obj = NULL;
+    // There should be a significant reward in order to tempt the player into
+    // trying to kill an RTP. Otherwise it's risk without reward.
+
+    // In the future this should be a random selection from a slew of different
+    // items ranging in usefulness.
+    obj = mksobj_at(MAGIC_MARKER, lev, x, y, TRUE, FALSE, rng);
```

Comment:
Is the Potter level the _only_ place an RTP will spawn? Otherwise this seems potentially game-breakingly powerful.

Reply:
I'm willing to discuss different item options. Perhaps we could make the magic marker have a very low drop chance?